stripe-core/src/main/java/com/stripe/android/core/utils/PluginDetector.kt

```diff
@@ -0,0 +1,36 @@
+package com.stripe.android.core.utils
+
+import androidx.annotation.RestrictTo
+
+/**
+ * A detector using reflection to check which plugin the SDK is being used from.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object PluginDetector {
+    /**
+     * Loop through all possible [PluginType]s and found the first one that exists in class path,
+     * return null if none of the class is found.
+     */
+    val pluginType: String? =
+        PluginType.values().firstOrNull {
+            isPlugin(it.className)
+        }?.pluginName
+
+    private fun isPlugin(className: String): Boolean = try {
+        Class.forName(className)
+        true
+    } catch (e: Exception) {
+        false
+    }
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    enum class PluginType(
+        val className: String,
+        val pluginName: String
+    ) {
+        ReactNative("com.facebook.proguard.annotations.DoNotStrip", "react-native"),
+        Flutter("io.flutter.embedding.engine.FlutterEngine", "flutter"),
+        Cordova("org.apache.cordova.CordovaActivity", "cordova"),
+        Unity("com.unity3d.player.UnityPlayerActivity", "unity")
```

Comment:
Looks good, any reason we don't have xamarin or ionic in this list?

Reply:
Wasn't able to find a Xamarin java class to test, will try investigate that in a follow up.
For ionic, per [this](https://stackoverflow.com/questions/51203863/access-android-native-code-from-ionic) thread, it seems it will be used alongside with Cordova, so will leave the Cordova enum to handle that

Reply:
If none can be found are we assuming that it is native? Is there a way to detect if the SDK is being used in a native app?

Reply:
Given that react+flutter occupies majority of the usecases and there might be new plugins introduced, I think it should be fine to assume this for now. If there is a dedicated new plugin that needs to be tracked we could always add them later